File: dotnet_3/cs/rest/SCS/Imp/FileSuffix.cs
namespace SCS.Imp
{
    public class FileSuffix
    {
        public static string Subject(string directory, string file)
        {
            var result = 0;

            //EG pathname = "...WORD/FILE.DOC";
            // files : Object[];
            //var lastFile : int = 0;

            var fileParts = file.Split(".");
            
            var lastPart = fileParts.Length - 1;
            
            if (lastPart <= 0) return "" + result;
            
            var suffix = fileParts[lastPart];

            if (directory == "text") {
                if ("txt".Equals(suffix)) {
                    result = 1;
                }
            }
            else if (directory == "acrobat") {
                if ("pdf".Equals(suffix)) {
                    //print("acrobat");
                    result = 2;
                }
            }
            else if (directory == "word") {
                if ("doc".Equals(suffix)) {
                    //print("word");
                    result = 3;
                }
            }
            else if (directory == "bin") {
                if ("exe".Equals(suffix)) {
                    //print("bin");
                    result = 4;
                }
            }
            else if (directory == "lib") {
                if ("dll".Equals(suffix)) {
                    //print("lib");
                    result = 5;
                }
            }

            return "" + result;
        }
    }
}

File: jdk_8_maven/cs/graphql/graphql-scs/src/main/kotlin/org/graphqlscs/type/FileSuffix.kt
package org.graphqlscs.type

import org.springframework.stereotype.Component

@Component
class FileSuffix {

    fun subject(directory: String, file: String): String {
        var result = 0

        //EG pathname = "...WORD/FILE.DOC";
        // files : Object[];
        var fileparts: Array<String?>? = null
        //var lastfile : int = 0;
        var lastpart = 0
        var suffix: String? = null
        fileparts = file.split(".".toRegex()).toTypedArray()
        lastpart = fileparts.size - 1
        if (lastpart > 0) {
            suffix = fileparts[lastpart]
            //Console.WriteLine("{0}, {1}", directory, suffix);
            if ("text" == directory) {
                if ("txt" == suffix) {
                    result = 1
                }
            }
            if ("acrobat" == directory) {
                if ("pdf" == suffix) {
                    //print("acrobat");
                    result = 2
                }
            }
            if ("word" == directory) {
                if ("doc" == suffix) {
                    //print("word");
                    result = 3
                }
            }
            if ("bin" == directory) {
                if ("exe" == suffix) {
                    //print("bin");
                    result = 4
                }
            }
            if ("lib" == directory) {
                if ("dll" == suffix) {
                    //print("lib");
                    result = 5
                }
            }
        }
        return "" + result
    }
}